docs/recipe_development.rst

```diff
@@ -0,0 +1,132 @@
+
+Registering a Recipe
+________
+
+
+In the case that a user already has a Python project set up with a structure similar to the layout below:
+
+This project can be found `here <https://github.com/justinGilmer/mbuild-fcc>`_.
+
+::
+
+    mbuild_fcc
+    ├── LICENSE
+    ├── README.md
+    ├── mbuild_fcc
+    │   ├── mbuild_fcc.py
+    │   └── tests
+    │       ├── __init__.py
+    │       └── test_fcc.py
+    └── setup.py
+
+
+The two important files for the user to convert their ``mBuild`` recipe to a discoverable plug-in are ``setup.py`` and ``mbuild_fcc.py``.
+
+To begin, lets first inspect the ``mbuild_fcc.py`` file, a shortened snippet is below.
+
+::
+
+    import mbuild
+
+
+    class FCC(mbuild.Compound):
+        """Create a mBuild Compound with a repeating unit of the FCC unit cell.
+
+        ... (shortened for viewability)
+
+        """
+
+        def __init__(self, lattice_spacing=None, compound_to_add=None, x=1, y=1, z=1):
+            super(FCC, self).__init__()
+
+            # ... (shortened for viewability)
+
+    if __name__ == "__main__":
+        au_fcc_lattice = FCC(lattice_spacing=0.40782,
+                             compound_to_add=mbuild.Compound(name="Au"),
+                             x=5, y=5, z=1)
+        print(au_fcc_lattice)
+
+
+
+There are two very important lines in this file that we need to know when developing a plug-in for mBuild.
+
+The first is the import statement ``import mbuild``.
+We must make sure that mbuild is installed since we are inheriting from ``mbuild.Compound``. When you decide to distribute your plug-in,
+the dependencies must be listed.
+
+The second is the name of the plug-in itself. It is considered good practice to name it the name of your ``class``.
+In this case, the name of the plug-in would be ``FCC``.
+
+The last step is to edit the ``setup.py`` file such that the plug-in can be registered under the entry_point group ``mbuild.plugins``.
+
+::
+
+    from setuptools import setup
+
+    setup(
+        name="mbuild_fcc",
+        install_requires="mbuild",
+        entry_points={
+            "mbuild.plugins":[
+                "FCC = mbuild_fcc.mbuild_fcc:FCC"
+            ]
+        },
+        py_modules=["mbuild_fcc"],
+    )
+
+This is once again, a very minimal setup file, a more thoroughly tested and developed package will have more information contained within.
+
+The important section is the ``entry_points`` argument. Here we define the entry_point group we want to register with: ``"mbuild.plugins"``.
+Finally, we tell python what name to use when accessing this plug-in, in this case we call it ``FCC`` as denoted by the name before the assignment operator ``FCC =``.
+Next, we pass the "path" from the location of the ``setup.py`` file to the python file that contains the class we want to register as a plug-in: ``mbuild_fcc.mbuild_fcc``. Then, we provide the name of the class within that python file we want to make discoverable ``:FCC``.
```

Comment:
`"path"`

I'm not sure what the proper term here is but I think this might confuse somebody to point it to the full path to the `.py` file